exchange/exchange_test.go

```diff
@@ -1196,6 +1201,131 @@ func TestCategoryDedupe(t *testing.T) {
 	assert.NotEqual(t, numIterations, selectedBids["bid_id3"], "Bid 3 made it through every time")
 }
 
+func TestBidRejectionErrors(t *testing.T) {
+	categoriesFetcher, error := newCategoryFetcher("./test/category-mapping")
+	if error != nil {
+		t.Errorf("Failed to create a category Fetcher: %v", error)
+	}
+
+	requestExt := newExtRequest()
+
+	targData := &targetData{
+		priceGranularity: requestExt.Prebid.Targeting.PriceGranularity,
+		includeWinners:   true,
+	}
+
+	requestExt.Prebid.Targeting.DurationRangeSec = []int{15, 30, 50}
+
+	adapterBids := make(map[openrtb_ext.BidderName]*pbsOrtbSeatBid)
+
+	cats := []string{}
+	bid := openrtb.Bid{ID: "bid_id1", ImpID: "imp_id1", Price: 10.0000, Cat: cats, W: 1, H: 1}
+	bid1_1 := pbsOrtbBid{&bid, "video", nil, &openrtb_ext.ExtBidPrebidVideo{Duration: 30}}
+
+	innerBids := []*pbsOrtbBid{
+		&bid1_1,
+	}
+
+	seatBid := pbsOrtbSeatBid{innerBids, "USD", nil, nil}
+	bidderName := openrtb_ext.BidderName("appnexus")
+
+	adapterBids[bidderName] = &seatBid
+
+	bidCategory, adapterBids, rejections, err := applyCategoryMapping(nil, requestExt, adapterBids, categoriesFetcher, targData)
```

Comment:
The test is confusing to read because it contains several different test cases. In this situation, it's ideal to use a table driven test approach. Look at the tests in the privacy package as examples.

Reply:
Updated the test to be more readable. I used a similar approach to the policy tests. Now runs a simple set of tests in a for loop so that it is easier to follow.